src/AltinnCore/Runtime/RestControllers/ValidateController.cs

```diff
@@ -0,0 +1,317 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Altinn.Platform.Storage.Models;
+
+using AltinnCore.Common.Configuration;
+using AltinnCore.Common.Helpers;
+using AltinnCore.Common.Services.Interfaces;
+using AltinnCore.Runtime.Models;
+using AltinnCore.Runtime.Validation;
+using AltinnCore.ServiceLibrary.Models;
+using AltinnCore.ServiceLibrary.Services.Interfaces;
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.Extensions.Options;
+
+namespace AltinnCore.Runtime.RestControllers
+{
+    /// <summary>
+    /// Represents all actions related to validation of data and instances
+    /// </summary>
+    [Authorize]
+    [ApiController]
+    public class ValidateController : ControllerBase
+    {
+        private readonly IData dataService;
+        private readonly IInstance instanceService;
+        private readonly IExecution executionService;
+        private readonly UserHelper userHelper;
+        private readonly IPlatformServices platformService;
+        private readonly IApplication appService;
+
+        /// <summary>
+        /// Initialises a new instance of the <see cref="ValidateController"/> class
+        /// </summary>
+        public ValidateController(
+            IOptions<GeneralSettings> generalSettings,
+            IRegister registerService,
+            IInstance instanceService,
+            IData dataService,
+            IExecution executionService,
+            IProfile profileService,
+            IPlatformServices platformService,
+            IInstanceEvent eventService,
+            IApplication appService)
+        {
+            this.instanceService = instanceService;
+            this.dataService = dataService;
+            this.executionService = executionService;
+            this.platformService = platformService;
+            this.appService = appService;
+            this.userHelper = new UserHelper(profileService, registerService, generalSettings);
+        }
+
+        /// <summary>
+        /// Validate an app instance. This will validate all individual data elements, both the binary elements and the elements bound
+        /// to a model, and then finally the state of the instance.
+        /// </summary>
+        /// <param name="org">Unique identifier of the organisation responsible for the app.</param>
+        /// <param name="app">Application identifier which is unique within an organisation</param>
+        /// <param name="instanceOwnerId">Unique id of the party that is the owner of the instance.</param>
+        /// <param name="instanceId">Unique id to identify the instance</param>
+        [Route("{org}/{app}/instances/{instanceOwnerId:int}/{instanceId:guid}/validate")]
+        public async Task<IActionResult> ValidateInstance(
```

Comment:
I was expecting that this method in fact call the validate method in the app so that any custom validation is done.

Reply:
Added a call to RunServiceEvent for validation for each data element with AppLogic true.